### ISSUE TITLE
[254-llm-chatbot] InferRequest.start_async() got an unexpected keyword argument 'shared_memory'

### ISSUE THREAD

**Issue by hsulin0806** [openvinotoolkit/openvino_notebooks#1592]
**Describe the bug**
I followed the following web installation process and found that whether I select CPU or GPU on Device, it will happen:

TypeError: InferRequest.start_async() got an unexpected keyword argument 'shared_memory'

![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/33116118/175940fc-045e-460b-8206-98cd44e83f32)

**Screenshots**
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/33116118/0f208da8-b01d-4a76-a4d2-df6a8ad85826)
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/33116118/f9df04bd-0c95-4845-acbe-96466e38baf0)

 **Installation instructions (Please mark the checkbox)**
[ ✓] I followed the installation guide at https://github.com/openvinotoolkit/openvino_notebooks#-installation-guide to install the notebooks.

**Comment by sanbuphy** [comment]
same

**Comment by hsulin0806** [comment]
Intel® Xeon(R) W-1290E CPU @ 3.50GHz × 20
Ubuntu 20.04.6 LTS
and
13th Gen Intel® Core™ i9-13900E × 32
Ubuntu 22.04.3 LTS
Both environments get the same result.

**Comment by OpenVINO-dev-contest** [comment]
Hi @hsulin0806 could you help to check the version of openvino in your system?
``` pip freeze | grep openvino```

**Comment by hsulin0806** [comment]
> 你好@hsulin0806你可以幫忙檢查一下你係統中openvino的版本嗎？ ` pip freeze | grep openvino`

Intel® Xeon(R) W-1290E CPU @ 3.50GHz × 20 
Ubuntu 20.04.6 LTS

pip freeze | grep openvino

nncf @ git+https://github.com/openvinotoolkit/nncf@dc928a9e45b2dba3fd6cf8ad8c046b56d5da35f3
openvino-nightly==2024.0.0.dev20240103
openvino-telemetry==2023.2.1
Note: you may need to restart the kernel to use updated packages.

and

13th Gen Intel® Core™ i9-13900E × 32
Ubuntu 22.04.3 LTS

pip freeze | grep openvino

openvino==2023.2.0
openvino-dev==2023.2.0
openvino-nightly==2024.0.0.dev20240103
openvino-telemetry==2023.2.1
Note: you may need to restart the kernel to use updated packages.

**Comment by OpenVINO-dev-contest** [comment]
could you uninstall ```openvino``` and ```openvino-dev```, then see if the error still exists ?

**Comment by hsulin0806** [comment]
> 您可以卸載`openvino`然後`openvino-dev`看看錯誤是否仍然存在？
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/33116118/f1ad16d2-442f-4de0-beb9-13149fe870b3)

An error will occur.

RuntimeError: Failed to import optimum.intel.openvino because of the following error (look up to see its traceback):
No module named 'openvino.runtime'

**Comment by OpenVINO-dev-contest** [comment]
> > 您可以卸載`openvino`然後`openvino-dev`看看錯誤是否仍然存在？
> > ![image](https://private-user-images.githubusercontent.com/33116118/294363727-f1ad16d2-442f-4de0-beb9-13149fe870b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MjAyOTQsIm5iZiI6MTcwNDQxOTk5NCwicGF0aCI6Ii8zMzExNjExOC8yOTQzNjM3MjctZjFhZDE2ZDItNDQyZi00ZGUwLWJlYjktMTMxNDlmZTg3MGIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA1VDAxNTk1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlNTc3ZTFiYzhmMGZkMGM1ODE0OTljZDJkNDJkZDQxM2U3ZGQxYmFmN2Y1MjNiOTA5ZTQ5YTUxNmYwYzMyZGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Y6OUcovxyY77M9-5FvkjSiDfW5-9HGRCyr2BBouanyc)
> 
> An error will occur.
> 
> RuntimeError: Failed to import optimum.intel.openvino because of the following error (look up to see its traceback): No module named 'openvino.runtime'

Just reinstalling ```optimum-intel``` shall be ok

**Comment by hsulin0806** [comment]
> > > 您可以卸載`openvino`然後`openvino-dev`看看錯誤是否仍然存在？
> > > ![image](https://private-user-images.githubusercontent.com/33116118/294363727-f1ad16d2-442f-4de0-beb9-13149fe870b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MjAyOTQsIm5iZiI6MTcwNDQxOTk5NCwicGF0aCI6Ii8zMzExNjExOC8yOTQzNjM3MjctZjFhZDE2ZDItNDQyZi00ZGUwLWJlYjktMTMxNDlmZTg3MGIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA1VDAxNTk1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlNTc3ZTFiYzhmMGZkMGM1ODE0OTljZDJkNDJkZDQxM2U3ZGQxYmFmN2Y1MjNiOTA5ZTQ5YTUxNmYwYzMyZGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Y6OUcovxyY77M9-5FvkjSiDfW5-9HGRCyr2BBouanyc)
> > 
> > 
> > An error will occur.
> > RuntimeError: Failed to import optimum.intel.openvino because of the following error (look up to see its traceback): No module named 'openvino.runtime'
> 
> Just reinstalling `optimum-intel` shall be ok
Again @@
TypeError: InferRequest.start_async() got an unexpected keyword argument 'shared_memory'

%pip freeze | grep openvino
nncf @ git+https://github.com/openvinotoolkit/nncf@6d6416d631497aaa2cf246c98ab31fb8c97d5847
openvino-nightly==2024.0.0.dev20240103
openvino-telemetry==2023.2.1
Note: you may need to restart the kernel to use updated packages.

**Comment by OpenVINO-dev-contest** [comment]
maybe you can try to install a lower version of openvino-nightly, .e.g. ```pip install openvino-nightly==2023.3.0.dev20231219

**Comment by hsulin0806** [comment]
> maybe you can try to install a lower version of openvino-nightly, .e.g. ```pip install openvino-nightly==2023.3.0.dev20231219
>
Thank u
pip install openvino-nightly==2023.3.0.dev20231219 This is the key!

The internal graphics is working, but the dGPU still cannot execute.

I use Intel Arc-370M dGPU
Kernel Restarting appears.
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/33116118/7541518a-5379-47a9-955f-6f96464930df)

**Comment by OpenVINO-dev-contest** [comment]
> > maybe you can try to install a lower version of openvino-nightly, .e.g. ```pip install openvino-nightly==2023.3.0.dev20231219
> 
> Thank u pip install openvino-nightly==2023.3.0.dev20231219 This is the key!
> 
> The internal graphics is working, but the dGPU still cannot execute.
> 
> I use Intel Arc-370M dGPU Kernel Restarting appears. ![image](https://private-user-images.githubusercontent.com/33116118/294384375-7541518a-5379-47a9-955f-6f96464930df.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MjQyMDEsIm5iZiI6MTcwNDQyMzkwMSwicGF0aCI6Ii8zMzExNjExOC8yOTQzODQzNzUtNzU0MTUxOGEtNTM3OS00N2E5LTk1NWYtNmY5NjQ2NDkzMGRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA1VDAzMDUwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI5YTcwMmIyNjI1MjkyMjk4NWExY2ZhOTFhMzIxNDNiNGIzY2JmMTA3NzFiYjZlZDgwN2NkMGRkMzgwZWRmZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lgjIAqXnfyem3CkvH9i04tTwAxJ1u72jv7wDTCJaeY8)

What is your model's size? a FP16/FP32 model exceed the maximum dedicated memory on ARC-370m, I believed its 4gGB ?

**Comment by hsulin0806** [comment]
> > > 也許你可以嘗試安裝較低版本的 openvino-nightly，例如 ```pip install openvino-nightly==2023.3.0.dev20231219
> > 
> > 
> > 謝謝 pip install openvino-nightly==2023.3.0.dev20231219 這才是關鍵！
> > 內部顯示卡正在工作，但 dGPU 仍然無法執行。
> > 我使用 Intel Arc-370M dGPU 出現核心重新啟動。![影像](https://private-user-images.githubusercontent.com/33116118/294384375-7541518a-5379-47a9-955f-6f96464930df.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MjQyMDEsIm5iZiI6MTcwNDQyMzkwMSwicGF0aCI6Ii8zMzExNjExOC8yOTQzODQzNzUtNzU0MTUxOGEtNTM3OS00N2E5LTk1NWYtNmY5NjQ2NDkzMGRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA1VDAzMDUwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI5YTcwMmIyNjI1MjkyMjk4NWExY2ZhOTFhMzIxNDNiNGIzY2JmMTA3NzFiYjZlZDgwN2NkMGRkMzgwZWRmZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lgjIAqXnfyem3CkvH9i04tTwAxJ1u72jv7wDTCJaeY8)
> 
> 您的模特兒尺寸是多少？FP16/FP32 型號可能超過 ARC-370m 上的最大專用內存，我相信它是 4gGB ？
>

Yes 4GB 64 bit Memory

You are right!
I tested the small model tiny-llama-1b-chat and it works normally, thank you!

**Comment by jiwaszki** [comment]
Hi, as the owner of most topics related to OV Python, here is the answer to the question: https://github.com/openvinotoolkit/openvino/issues/21979#issuecomment-1878154656

tl;dr you need to replace `shared_memory` with `share_inputs`. It works the same. I also encourage you to try out new `share_outputs` flag, especially with LLMs!

Cheers!

**Comment by hsulin0806** [comment]
> Hi, as the owner of most topics related to OV Python, here is the answer to the question: [openvinotoolkit/openvino#21979 (comment)](https://github.com/openvinotoolkit/openvino/issues/21979#issuecomment-1878154656)
> 
> tl;dr you need to replace `shared_memory` with `share_inputs`. It works the same. I also encourage you to try out new `share_outputs` flag, especially with LLMs!
> 
> Cheers!

Thank you! I will try 'share_outputs'.
May I ask, does the '254-llm-chatbot' example provide a C++ inference program like the object recognition example?

**Comment by jiwaszki** [comment]
@hsulin0806 you may start by looking at https://github.com/openvinotoolkit/openvino.genai/tree/master/text_generation/causal_lm/cpp

**Comment by hsulin0806** [comment]
> @hsulin0806 you may start by looking at https://github.com/openvinotoolkit/openvino.genai/tree/master/text_generation/causal_lm/cpp

Thank you!
